src/shogun/base/AnyParameter.h

```diff
@@ -37,7 +37,8 @@ namespace shogun
 		HYPER = 1u << 0,
 		GRADIENT = 1u << 1,
 		MODEL = 1u << 2,
-		AUTO = 1u << 10
+		AUTO = 1u << 10,
+		CONST = 1u << 11
```

Comment:
`READ_ONLY` might be a better name to avoid clashes

Reply:
yup, I didn't realise that it could clash with C++ keywords...